tests/unit-tests/internal/student-progress/course-progress/repositories/test-class-tables-based-course-progress-repository.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+namespace SenseiTest\Internal\Student_Progress\Course_Progress\Repositories;
+
+use DateTimeImmutable;
+use Sensei\Internal\Student_Progress\Course_Progress\Models\Course_Progress;
+use Sensei\Internal\Student_Progress\Course_Progress\Repositories\Tables_Based_Course_Progress_Repository;
+use wpdb;
+
+/**
+ * Tests for Tables_Based_Course_Progress_Repository.
+ *
+ * @covers \Sensei\Internal\Student_Progress\Course_Progress\Repositories\Tables_Based_Course_Progress_Repository
+ */
+class Tables_Based_Course_Progress_Repository_Test extends \WP_UnitTestCase {
+
+	public function testCreate_ParamsGiven_InsertsToWpdb(): void {
+		/* Arrange. */
+		$wpdb       = $this->createMock( wpdb::class );
+		$repository = new Tables_Based_Course_Progress_Repository( $wpdb );
+
+		/* Expect & Act. */
+		$wpdb
+			->expects( $this->once() )
+			->method( 'insert' )
+			->with(
+				'sensei_lms_progress',
+				$this->callback(
+					function( $array ) {
+						if ( ! isset( $array['post_id'] ) || 1 !== $array['post_id'] ) {
+							return false;
+						}
+						if ( ! isset( $array['user_id'] ) || 2 !== $array['user_id'] ) {
+							return false;
+						}
+						if ( ! array_key_exists( 'parent_post_id', $array ) || ! is_null( $array['parent_post_id'] ) ) {
+							return false;
+						}
+						if ( ! isset( $array['type'] ) || 'course' !== $array['type'] ) {
+							return false;
+						}
+						if ( ! isset( $array['status'] ) || 'in-progress' !== $array['status'] ) {
+							return false;
+						}
+						return true;
```

Comment:
I found this part a bit hard to follow. What do you think about the following suggestion?

```suggestion
					function( $array ) {
						return 1 === $array['post_id']
							&& 2 === $array['user_id']
							&& is_null( $array['parent_post_id'] )
							&& 'course' === $array['type']
							&& 'in-progress' === $array['status'];
					}
```

Reply:
This applies to a few more places as well.

Reply:
Okay, I see. Your suggestion is much better in terms of readability. However, it is not good to omit the `isset` check. I'll change slightly your suggestion by checking keys.

Reply:
Fixed here: https://github.com/Automattic/sensei/pull/5894/commits/69f3c0dcc492c18fa00d2ba4ef6a14f89140753f